core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorConcurrencyTest.scala

```diff
@@ -106,25 +97,56 @@ class GroupCoordinatorConcurrencyTest extends AbstractCoordinatorConcurrencyTest
     }
   }
 
-  def createGroupMembers(groupPrefix: String): Set[GroupMember] = {
-    (0 until nGroups).flatMap { i =>
-      new Group(s"$groupPrefix$i", nMembersPerGroup, groupCoordinator, replicaManager).members
-    }.toSet
+  private var producerIdCount = 0L
+  private val allGroupMembers = mutable.ArrayBuffer[GroupMember]()
+
+  def groupMembers(groupId: String, nMembers: Int, groupCoordinator: GroupCoordinator): Seq[GroupMember] = {
+    val groupPartitionId = groupCoordinator.partitionFor(groupId)
+    groupCoordinator.groupManager.addPartitionOwnership(groupPartitionId)
+    val lock = new ReentrantLock()
+    val producerId = producerIdCount
+    producerIdCount += 1
```

Comment:
I think the intention for the test is probably to use the same producerId since it tests more on transactional conflicts.

Reply:
Got it. However, the same producerId means the group completed by CompleteTxnOperation is possible to be impacted by any CommitTxnOffsetsOperation (since the partitions are same also). Hence, the side-effect is that we need a single lock to control the happen-before of txn completion and commit so the test will get slower.